bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,873 @@
+        }
+
+        @Override
+        public boolean onIdleTimeout(@Nullable Session session) {
+            return false;
+        }
+
+        @Override
+        public void onReset(@Nullable Session session, @Nullable ResetFrame frame) {
+            fatalError(Error.RESET);
+        }
+    }
+
+    private static final String APPLICATION_ID = "org-openhab-binding-hue-clip2";
+    private static final String APPLICATION_KEY = "hue-application-key";
+
+    private static final String FORMAT_URL_CONFIG = "http://%s/api/0/config";
+    private static final String FORMAT_URL_RESOURCE = "https://%s/clip/v2/resource/";
+    private static final String FORMAT_URL_REGISTER = "http://%s/api";
+    private static final String FORMAT_URL_EVENTS = "https://%s/eventstream/clip/v2";
+
+    private static final int CLIP2_MINIMUM_VERSION = 1948086000;
+    private static final int TIMEOUT_SECONDS = 10;
+    private static final int CHECK_ALIVE_SECONDS = 300;
+
+    private static final ResourceReference BRIDGE = new ResourceReference().setType(ResourceType.BRIDGE);
+
+    /**
+     * Static method to attempt to connect to a Hue Bridge, get its software version, and check if it is high enough to
+     * support the CLIP 2 API.
+     *
+     * @param hostName the bridge IP address.
+     * @return returns without any exception if the bridge is online and it does support CLIP 2.
+     * @throws ApiException if was not possible to connect or another error was encountered.
+     * @throws IllegalArgumentException if the hostName is bad.
+     * @throws IllegalStateException if it was possible to connect but CLIP 2 is not supported.
+     */
+    public static void testSupportsClip2(String hostName)
+            throws ApiException, IllegalStateException, IllegalArgumentException {
+        //
+        HttpRequest request;
+        try {
+            request = HttpRequest.newBuilder().uri(new URI(String.format(FORMAT_URL_CONFIG, hostName)))
+                    .header(HttpHeader.ACCEPT.toString(), MediaType.APPLICATION_JSON)
+                    .timeout(Duration.ofSeconds(TIMEOUT_SECONDS)).build();
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Bad host name");
+        }
+        HttpResponse<String> response;
+        try {
+            response = java.net.http.HttpClient.newHttpClient().send(request, BodyHandlers.ofString());
+        } catch (IOException | InterruptedException e) {
+            throw new ApiException("Communication error", e);
+        }
+        if (response.statusCode() == 200) {
+            BridgeConfig config = new Gson().fromJson(response.body(), BridgeConfig.class);
+            if (Objects.nonNull(config)) {
+                String swVersion = config.swversion;
+                if (Objects.nonNull(swVersion)) {
+                    if (Integer.parseInt(swVersion) < CLIP2_MINIMUM_VERSION) {
+                        throw new IllegalStateException("Hue Bridge does not support CLIP 2");
```

Comment:
This exception type does not seem appropriate here: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/IllegalStateException.html

Isn't this the normal outcome rather than something exceptional? It seems to me the method could return `boolean` and be renamed to something like `isClip2Supported`.

Reply:
> It seems to me the method could return boolean

No. There is not just one boolean outcome, there are four possible outcomes (normal return, plus three different possible exceptions) which the handler (code below) treats differently. (To return more than just a boolean, one could return a String, an Enum, or an Int, but I think that those would also require more explanation than they are worth..)

![image](https://user-images.githubusercontent.com/893994/211362847-1ebf76cd-614a-4799-978f-1faee86c81c3.png)

> exception type does not seem appropriate here:

Do you have another suggestion?

Reply:
You have three outcomes:
- API v2 is supported.
- API v2 is not supported.
- You are not able to determine if API v2 is supported.

If you are not able to determine this, that justifies throwing an exception, because that's something unexpected. Besides that, it seems faster an simpler to return `true` if supported and `false` if not.

So my suggestion would be to return `true` in the line where you now just return, and `false` in the line where you now throw `IllegalStateException`.

Reply:
Done.